contracts/ChipRegistry.sol

```diff
@@ -137,225 +124,224 @@ contract ChipRegistry is IChipRegistry, ClaimedPBT, Ownable {
-
-        projectEnrollments[_projectRegistrar].merkleRoot = _merkleRoot;
-        projectEnrollments[_projectRegistrar].projectClaimDataUri = _projectClaimDataUri;
-        
-        emit ProjectMerkleRootUpdated(address(_projectRegistrar), _merkleRoot, _projectClaimDataUri);
-    }
-
-    /**
-     * @notice Allow a user to claim a chip from a project enrollment. Enrollment allows the chip to resolve to the project's preferred
+     * @notice Allow a project to add chips. Enrollment allows the chip to resolve to the project's preferred
      * service. Additionally, claiming creates a Physically-Bound Token representation of the chip.
      *
-     * @dev This function will revert if the chip has already been claimed, if invalid certificate data is provided or if the chip is
+     * @dev This function will revert if the chip has already been added, if invalid certificate data is provided or if the chip is
      * not part of the project enrollment (not in the project merkle root). Addtionally, there are checks to ensure that the calling
      * ProjectRegistrar has implemented the correct ERS logic. This function is EIP-1271 compatible and can be used to verify chip
      * claims tied to an account contract.
      *
      * @param _chipId                       Chip ID (address)
-     * @param _chipClaim                    Struct containing information for validating merkle proof, chip owner, and chip's ERS node
+     * @param _chipOwner                    Struct containing information for validating merkle proof, chip owner, and chip's ERS node
+     * @param _nameHash                     Label of the node in the ERS tree; typically the chipId unless the project wishes to use 
+     *                                      another unique identifier. The full ersNode will be used as the tokenId for the issued PBT.
      * @param _manufacturerValidation       Struct containing information for chip's inclusion in manufacturer's merkle tree
-     * @param _developerInclusionProof      Signature of the chipId signed by the project's public key
-     * @param _developerCustodyProof        Signature of the projectPublicKey signed by the chip's private key
      */
-    function claimChip(
+    
+    function addChip(
         address _chipId,
-        ChipClaim calldata _chipClaim,
-        ManufacturerValidation memory _manufacturerValidation,
-        bytes memory _developerInclusionProof,
-        bytes memory _developerCustodyProof
+        address _chipOwner,
+        bytes32 _nameHash,
+        IChipRegistry.ManufacturerValidation memory _manufacturerValidation
     )
-        external virtual
+        external
+        virtual
     {
-        ProjectInfo memory projectInfo = projectEnrollments[IProjectRegistrar(msg.sender)];
+        IProjectRegistrar projectRegistrar = IProjectRegistrar(msg.sender);
+        ProjectInfo memory projectInfo = projectEnrollments[projectRegistrar];
```

Comment:
This needs to be permissioned in some way. Maybe check to make sure that the project registrar has been added? Right now anyone could add chips under a projects `rootNode` simply by saying that their `projectRegistrar` `rootNode` is another project's `rootNode`

Reply:
I believe we are set here with WIP commit since we are no longer calling projectRegistrar for subsequent rootNode checks.